src/lib/features/authenticate/model/track-auth-status.ts

```diff
@@ -0,0 +1,15 @@
+import noop from 'lodash.noop';
+import { user } from '$lib/entities/user';
+import { supabaseClient } from '$lib/shared/api';
+
+export function trackAuthStatus() {
+  const { data, error } = supabaseClient.auth.onAuthStateChange((_event, session) => {
+    user.set(session?.user ?? null);
+  });
+
+  if (error !== null) {
+    throw error;
+  }
+
+  return data?.unsubscribe ?? noop;
```

Comment:
Question: I see `noop` is used only as the funtion returned from the `trackAuthStatus` function, but the return data is never actually used in the code. So, do we need to have `noop` intalled as a dependency if it isn't used?

Reply:
It is used — the function that you return from an `onMount` callback will be called on unmounting to clean up. And this dependency is so tiny I don't think it's ever going to be an issue.

It's just that we have ESLint set up to forbid empty functions (which is good, if I think about it)